### ISSUE TITLE
KeyError: 'sections'

### ISSUE THREAD

**Issue by pawamoy** [frostming/pdm#257]
- [x] I have searched the issue tracker and believe that this is not a duplicate.

## Steps to reproduce

```console
$ pdm sync -dv
Traceback (most recent call last):
  File "/home/user/.local/bin/pdm", line 8, in <module>
    sys.exit(main())
  File "/home/user/.local/pipx/venvs/pdm/lib/python3.8/site-packages/pdm/core.py", line 73, in __call__
    return self.main(*args, **kwargs)
  File "/home/user/.local/pipx/venvs/pdm/lib/python3.8/site-packages/pdm/core.py", line 112, in main
    raise err.with_traceback(traceback)
  File "/home/user/.local/pipx/venvs/pdm/lib/python3.8/site-packages/pdm/core.py", line 108, in main
    f(options.project, options)
  File "/home/user/.local/pipx/venvs/pdm/lib/python3.8/site-packages/pdm/cli/commands/sync.py", line 18, in handle
    actions.do_sync(
  File "/home/user/.local/pipx/venvs/pdm/lib/python3.8/site-packages/pdm/cli/actions.py", line 121, in do_sync
    candidates.update(project.get_locked_candidates("dev"))
  File "/home/user/.local/pipx/venvs/pdm/lib/python3.8/site-packages/pdm/project/core.py", line 315, in get_locked_candidates
    if section != "__all__" and section not in package["sections"]:
KeyError: 'sections'
```

## Actual behavior

It seems that when locking some dependencies are not given any section (there is no "sections" key in pdm.lock):

```
[[package]]
name = "future"
version = "0.18.2"
summary = "Clean single-source support for Python 3 and 2"

...

[[package]]
name = "joblib"
version = "1.0.0"
summary = "Lightweight pipelining with Python functions"

...

[[package]]
name = "lunr"
version = "0.5.8"
summary = "A Python implementation of Lunr.js"
dependencies = [
    "future>=0.16.0",
    "six>=1.11.0",
]

...

[[package]]
name = "nltk"
version = "3.5"
marker = "python_version >= '2.8'"
summary = "Natural Language Toolkit"
dependencies = [
    "click",
    "joblib",
    "regex",
    "tqdm",
]
```

## Expected behavior

No errors!

## Environment Information

```console
$ pdm info && pdm info --env
PDM version:        1.3.0                                                                 
Python Interpreter: /home/user/.basher-packages/pyenv/pyenv/shims/python (3.6.12)
Project Root:       /home/user/data/dev/project
{
  "implementation_name": "cpython",
  "implementation_version": "3.6.12",
  "os_name": "posix",
  "platform_machine": "x86_64",
  "platform_release": "3.10.0-862.6.3.el7.x86_64",
  "platform_system": "Linux",
  "platform_version": "#1 SMP Fri Jun 15 17:57:37 EDT 2018",
  "python_full_version": "3.6.12",
  "platform_python_implementaiton": "CPython",
  "python_version": "3.6",
  "sys_platform": "linux"
}
```

**Comment by frostming** [comment]
You have to regenerate the lockfile, it was due to a bug fixed on 1.3.0 You have to regenerate the lockfile, it was due to a bug fixed on 1.3.0

**Comment by pawamoy** [comment]
Dang I thought I did, but apparently not. I deleted pdm.lock, relocked, and the KeyError 'sections' does not appear anymore. Lets consider this fixed. Thanks a lot!
